README.md

```diff
@@ -1,170 +1,189 @@
+  - **API Docs (Swagger)**: `http://localhost:8000/docs`
+
+**Option B: With Kubernetes with Tilt**
+
+This method is ideal for simulating a production environment.
 
-#### Option 3: Kubernetes with Tilt
 ```bash
-# From project root:
+# Ensure your local Kubernetes cluster is running, then start tilt
 tilt up
-
-# The app will be available at http://localhost:10005
-# Tilt UI available at http://localhost:10350
 ```
+The application will be available at `http://localhost:10005`, with the Tilt UI at `http://localhost:10350`.
 
-### 3. Code Quality
 
-```bash
-uv run poe check      # Run all quality checks
-uv run poe mypy       # Type checking
-uv run poe ruff       # Linting and formatting
-uv run poe bandit     # Security analysis
-```
+-----
+
+## Configuration
+
+Current configurations are managed within the source files.
+
+Key configuration points include:
 
-## Local Kubernetes Setup
+  * **Server Host/Port**: Defined in the FastAPI application startup logic in `src/agent_monitor/main.py`.
+  * **Logging**: Configured in `src/agent_monitor/main.py`, with health check endpoints excluded from access logs.
+  * **OTLP Span Processing**: Business logic is located in `src/agent_monitor/core/span_preprocessor.py`.
 
-For local development, you need a Kubernetes cluster.
-Use your preferred method to set up a local Kubernetes cluster. Docker Desktop, Ranger Desktop and Colima all provide standard Kubernetes clusters. Those are recommended.
+-----
 
-Alternatively, use `k3d` for a lightweight solution.
+## End-to-End (E2E) Testing
 
-Using `k3d`, create a local registry and cluster:
+The project includes a comprehensive test suite to ensure reliability and correctness.
+
+### Prerequisites for Testing
+
+  * All Python dependencies must be installed via `uv sync`.
+  * The tests utilize mock span data located at `tests/mock_spans.json`. No running database or external services are required.
+
+### Running the Test Suite
+
+Use the following commands to execute the tests:
+
+```bash
+# Run the full test suite once
+uv run poe test
+```
 ```bash
-brew install k3d
-k3d registry create local-paal-registry --port 6169
-# Currently required for Colima users. See https://github.com/k3d-io/k3d/pull/1584
-export K3D_FIX_DNS=0
-k3d cluster create local-paal --registry-use k3d-local-paal-registry
+# Run tests in watch mode for continuous testing during development
+uv run poe test-watch
+```
+```bash
+# Run a specific test file with verbose output
+uv run pytest tests/test_integration.py -v
+```
+```bash
+# Run tests with coverage analysis
+uv run pytest --cov=src/agent_monitor
 ```
 
-## API Reference
+The integration tests validate the complete workflow, from OTLP trace ingestion to WebSocket event broadcasting.
 
-### Trace Ingestion
+-----
 
-- **POST** `/v1/traces` - Receives OpenTelemetry trace data (protobuf format) following the [OTLP specification](https://opentelemetry.io/docs/specs/otlp/#otlphttp-request)
+## Testing Tools and Their Configuration
```

Comment:
Not sure about this section. Is it really needed? Can easily be looked up for details.